nncf/quantization/quantize.py

```diff
@@ -78,3 +81,53 @@ def quantize(model: TModel,
+                                   validation_dataset: Dataset,
+                                   validation_fn: Callable[[Any, Iterable[Any]], float],
+                                   max_drop: float = 0.01,
+                                   preset: QuantizationPreset = QuantizationPreset.PERFORMANCE,
+                                   target_device: TargetDevice = TargetDevice.ANY,
+                                   subset_size: int = 300,
+                                   fast_error_correction: bool = True,
+                                   model_type: Optional[str] = None,
+                                   ignored_scope: Optional[IgnoredScope] = None) -> ModelType:
+    """
+    Applies post-training quantization algorithm with accuracy control to provided model.
+
+    :param model: A model to be quantized.
+    :param calibration_dataset: A representative dataset for the calibration process.
+    :param validation_dataset: A dataset for the validation process.
+    :param validation_fn: A validation function to validate the model. It should take
+        two argumets:
+        - `model`: model to be validate.
+        - `validation_dataset`: dataset that provides data items to
+              validate the provided model.
+        The function should return the value of the metric with the following meaning:
+        A higher value corresponds to better performance of the model.
+    :param max_drop: The maximum accuracy drop that should be achieved after the quantization.
+    :param preset: A preset that controls the quantization mode.
+    :param target_device: A target device the specificity of which will be taken
+        into account while compressing in order to obtain the best performance
+        for this type of device.
+    :param subset_size: Size of a subset to calculate activations
+        statistics used for quantization.
+    :param fast_error_correction: Setting this option to `False` enables a different
+        bias correction method which is more accurate, in general, and takes
+        more time but requires less memory.
+    :param model_type: Model type is needed to specify additional patterns
+        in the model. Supported only `transformer` now.
+    :param ignored_scope: An ignored scope that defined the list of model control
+        flow graph nodes to be ignored during quantization.
+    :return: The quantized model.
+    """
+    backend = get_backend(model)
+    if backend == BackendType.OPENVINO:
+        from nncf.openvino.quantization.quantize import quantize_with_accuracy_control_impl
+        return quantize_with_accuracy_control_impl(model, calibration_dataset, validation_dataset, validation_fn,
+                                                   max_drop, preset, target_device, subset_size,
+                                                   fast_error_correction, model_type, ignored_scope)
+
+    raise RuntimeError(f'Unsupported type of backend: {backend}')
```

Comment:
How about using `NotImplementedError` instead of `RuntimeError`?

Reply:
Good question! I think using `RuntimeError` is preferable here then `NotImplementedError`. Please see [here](https://docs.python.org/3/library/exceptions.html#NotImplementedError) for more details